native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/listener/ServerConnectorBootstrap.java

```diff
@@ -297,7 +294,9 @@ private boolean unBindInterface() throws InterruptedException {
             ChannelFuture future = getChannelFuture();
             if (future != null) {
                 //Close will stop accepting new connections.
-                future.channel().close().sync();
+                ChannelFuture closeFuture = future.channel().close();
+                Thread.sleep(gracefulStopTimeout);
+                closeFuture.sync();
```

Comment:
I thought the code should be

Thread.sleep(gracefulStopTimeout);
future.channel().close().sync();

Reply:
If the sleep was executed before the close, unbind does not happen until the close is called. So it's just a thread block.

This grace period is intend to count after executing close() to provide some time to cater already accepted request. 
closeFuture.sync() is called afterward to wait till the close is succeeded.